inc/signup_notification/namespace.php

```diff
@@ -160,3 +185,340 @@ function altis_signup_blog_notification( $domain, $path, $title, $user_login, $u
 
 	return true;
 }
+
+/**
+ * Sends a confirmation request email to a user when they sign up for a new user account (without signing up for a site
+ * at the same time). The user account will not become active until the confirmation link is clicked.
+ *
+ * This is the notification function used when no new site has
+ * been requested.
+ *
+ * Filter {@see 'wpmu_signup_user_notification'} to bypass this function or
+ * replace it with your own notification behavior.
+ *
+ * Filter {@see 'wpmu_signup_user_notification_email'} and
+ * {@see 'wpmu_signup_user_notification_subject'} to change the content
+ * and subject line of the email sent to newly registered users.
+ *
+ * @since MU (3.0.0)
+ *
+ * @param string $user_login The user's login name.
+ * @param string $user_email The user's email address.
+ * @param string $key        The activation key created in wpmu_signup_user()
+ * @param array  $meta       Optional. Signup meta data. Default empty array.
+ * @return bool
+ */
+function altis_signup_user_notification( $user_login, $user_email, $key, $meta = [] ) {
+	/**
+	 * Filters whether to bypass the email notification for new user sign-up.
+	 *
+	 * @since MU (3.0.0)
+	 *
+	 * @param string $user_login User login name.
+	 * @param string $user_email User email address.
+	 * @param string $key        Activation key created in wpmu_signup_user().
+	 * @param array  $meta       Signup meta data. Default empty array.
+	 */
+	if ( ! apply_filters( 'wpmu_signup_user_notification', $user_login, $user_email, $key, $meta ) ) {
+		return false;
+	}
+
+	// Patch error in wp-user-signups passing through signup key.
```

Comment:
It's not clear why this (and `altis_signup_blog_notification`) might be called with a blank `$key`. The WP core original doesn't seem to cater for that case. The comment mentions `wp-user-signups` but is that a function, a filter?

Reply:
WP User Signups is a plugin this module provides to expose a UI for signups in the network admin. There’s a bug in that plugin which means the  key can come through empty.